featuretools/entityset/entityset.py

```diff
@@ -570,6 +554,8 @@ def entity_from_dataframe(self,
             make_index=make_index)
         self.entity_dict[entity.id] = entity
         self.reset_data_description()
+
```

Comment:
entity_from_dataframe should probably have `_check_time_index` in it.  It makes sense to validate as we add an entity.  This would probably remove some of the stray `_check_time_index` calls being added to tests

Reply:
We can add in `entity_from_dataframe`. I wasn't sure if we were working towards limiting the time type check to only happen in CFM.